contracts/DuskMainnetOnramp.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MPL-2.0
+pragma solidity ^0.8.24;
+
+// Minimal ERC20 interface required to transfer DUSK
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    ) external returns (bool);
+}
+
+/// @title The ERC20/BEP20 DUSK mainnet on-ramp contract
+/// @author Hein Dauven
+contract DuskMainnetOnramp {
+    IERC20 public immutable duskToken;
+    // Conversion factor between ERC20/BEP20 DUSK (18 decimals) and native DUSK (9 decimals), where 10^9 DUSK wei is equivalent to 1 LUX
+    uint256 constant LUX_CONVERSION_FACTOR = 10**9; 
+    uint256 constant MINIMAL_STAKE = 1000 * 10**9 * LUX_CONVERSION_FACTOR;
```

Comment:
nit, but I would like to see explicit parentheses used to directly see the precedence of operators like in other parts of the code.